### PR TITLE
 shipit_uplift: Add API endpoint to generate summary of coverage information for a given commit

### PR DESCRIPTION
**Pull request by marco-c**
This depends on #595. I will rebase once that one is merged.

CC @armenzg

### CODE REVIEW

**Comment by marco-c** [comment]
@armenzg r?

File: src/shipit_uplift/shipit_uplift/api.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
     serialize_patch_status
 )
 from shipit_uplift.config import SCOPES_USER, SCOPES_BOT, SCOPES_ADMIN
-from shipit_uplift import coverage_by_dir_impl, coverage_by_changeset_impl
+from shipit_uplift import coverage_by_dir_impl, coverage_by_changeset_impl, coverage_summary_by_changeset_impl
```
Comment (armenzg) [inline src/shipit_uplift/shipit_uplift/api.py]: Could you please use the following style?

```python
from x import (
    a,
    b
)

```

File: src/shipit_uplift/shipit_uplift/coverage.py [hunk 2]
```diff
@@ -42,9 +42,19 @@ class CoverallsCoverage(Coverage):
     URL = 'https://coveralls.io'
 
     @staticmethod
-    def has_coverage(changeset):
+    @cache.memoize()
+    def get_coverage(changeset):
```
Comment (armenzg) [inline src/shipit_uplift/shipit_uplift/coverage.py]: Caching! woohoo!

File: src/shipit_uplift/shipit_uplift/api.py (hunk 1 above)
Comment (marco-c) [reply src/shipit_uplift/shipit_uplift/api.py]: Done!

**Comment by marco-c** [comment]
@jankeromnes this is like the other PR (#595), mostly containing only coverage API endpoint changes. Could you approve it so it can be merged?

**Pull request merged**